src/pools/selection-strategies.ts

```diff
@@ -0,0 +1,112 @@
+import type { JSONValue } from '../utility-types'
+import { isKillBehavior, KillBehaviors } from '../worker/worker-options'
+import type { IWorker } from './abstract-pool'
+import type { IDynamicPool } from './dynamic-pool'
+
+/**
+ * Result of the round robin selection function.
+ *
+ * @template Type of the chosen element.
+ */
+export interface RoundRobinSelectionResult<Element> {
+  /** The element that was chosen. */
+  chosenElement: Element
+  /** The next calculated index. */
+  nextIndex: number
+}
+
+/**
+ * Selects the next element in a round robin selection based on the given index.
+ *
+ * @template Element Type of the element.
+ * @param elements An array of elements.
+ * @param nextIndex The next calculated index.
+ * @returns The chosen element together with the next calculated index.
+ */
+export function roundRobinSelection<Element> (
+  elements: readonly Element[],
+  nextIndex: number
+): RoundRobinSelectionResult<Element> {
+  const chosenElement = elements[nextIndex]
+  nextIndex = elements.length - 1 === nextIndex ? 0 : nextIndex + 1
+  return { chosenElement, nextIndex }
+}
```

Comment:
@pioardi @jerome-benoit 

We could replace this with
```ts
export function roundRobinSelection<Worker extends IWorker, Data, Response> (
  poolReference: AbstractPool<Worker, Data, Response>
): Worker {
  const chosenWorker = poolReference.workers[poolReference.nextWorkerIndex]
  poolReference.nextWorkerIndex =
    poolReference.workers.length - 1 === poolReference.nextWorkerIndex
      ? 0
      : poolReference.nextWorkerIndex + 1
  return chosenWorker
}
```

That way we can pass the pool directly and can remove the `RoundRobinSelectionResult` and so on.
The only downside of that would be that it is not unit-testable on its own and rely on the pool. But that would be okay for me.

What do you think?

Reply:
It's too complicated for one strategy implementation and not the same as the other strategy. The have to share a common signature to allow to add easily more later.